app/models/grda_warehouse/enrollment_change_history.rb

```diff
@@ -42,17 +43,20 @@ def self.create_for_clients_on_date! client_ids:, date:
     end
 
     def self.attributes_for_client_on_date client:, date:
+      now = Time.current
       attributes_for_client = {
         client_id: client.id,
         on: date,
         # are timestamps and version columns used? This table is large; perhaps they could be dropped for efficiency
-        created_at: Time.now,
-        updated_at: Time.now,
+        created_at: now,
+        updated_at: now,
         version: 1,
       }
-      attributes_for_client[:residential] = client.enrollments_for_rollup(en_scope: client.scope_for_residential_enrollments(current_user), user: User.setup_system_user).to_json rescue '[]' # rubocop:disable Style/RescueModifier
-      attributes_for_client[:other] = client.enrollments_for_rollup(en_scope: client.scope_for_other_enrollments(current_user), user: User.setup_system_user).to_json rescue '[]' # rubocop:disable Style/RescueModifier
-      attributes_for_client[:days_homeless] = client.days_homeless rescue 0 # rubocop:disable Style/RescueModifier
+
+      user = User.setup_system_user
```

Comment:
If I understand correctly, using the system user means the enrollment history will include all enrollments. Previously this used current_user which suggests the intent to restrict visibility. Does this seem okay?

Reply:
Correct, and yes, System User is correct.  Prior to adding the `current_user` that caused it to fail, it wasn't including a user at all.  This is behind an explicit permission and is used to troubleshoot historic problems with data loading.  It allows someone to answer the questions: "they had an enrollment here last month, but now it's gone? or this enrollment had 300 days of service and now only has 10, why?"  We use this log to determine when the change occurred and if it was gradual or sudden.  Once we've identified when, we can dig through import logs or paper trail to figure out the cause if necessary.

Reply:
thanks, i'll add that to the inline docs